components/automate-cli/cmd/chef-automate/automateHAAwsDeployment.go

```diff
@@ -252,59 +272,173 @@ func (a *awsDeployment) validateEnvFields() *list.List {
+				len(strings.TrimSpace(a.config.ChefServer.Config.PublicKey)) < 1 {
+				errorList.PushBack("ChefServer root_ca and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false.")
+			}
+			errorList.PushBackList(checkCertValid([]keydetails{
+				{key: a.config.ChefServer.Config.PrivateKey, certtype: "private_key", svc: "chef-server"},
+				{key: a.config.ChefServer.Config.PublicKey, certtype: "public_key", svc: "chef-server"},
+			}))
 		}
-		errorList.PushBackList(checkCertValid([]keydetails{
-			{key: a.config.ChefServer.Config.PrivateKey, certtype: "private_key", svc: "chef-server"},
-			{key: a.config.ChefServer.Config.PublicKey, certtype: "public_key", svc: "chef-server"},
-		}))
 	}
+
 	if a.config.Postgresql.Config.EnableCustomCerts {
-		if len(strings.TrimSpace(a.config.Postgresql.Config.RootCA)) < 1 ||
-			len(strings.TrimSpace(a.config.Postgresql.Config.PrivateKey)) < 1 ||
-			len(strings.TrimSpace(a.config.Postgresql.Config.PublicKey)) < 1 {
-			errorList.PushBack("Postgresql root_ca and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false.")
+		if len(a.config.Postgresql.Config.CertsByIP) > 0 {
+			if len(strings.TrimSpace(a.config.Postgresql.Config.RootCA)) < 1 {
+				errorList.PushBack("Postgresql root_ca is missing. Set custom_certs_enabled to false to continue without custom certificates.")
+			}
+			errorList.PushBackList(checkCertValid([]keydetails{
+				{key: a.config.Postgresql.Config.RootCA, certtype: "root_ca", svc: "postgresql"},
+			}))
+			if !stringutils.SubSlice(a.configPostgresqlIpList, extractIPsFromCertsByIP(a.config.Postgresql.Config.CertsByIP)) {
+				errorList.PushBack("Missing certificates for some Postgresql private ips. Please make sure certificates for the following ips are provided in certs_by_ip: " + strings.Join(a.configPostgresqlIpList, ", "))
+			}
+			// check if all the certs are valid for given IPs
+			for _, node := range a.config.Postgresql.Config.CertsByIP {
+				if len(strings.TrimSpace(node.IP)) < 1 ||
+					len(strings.TrimSpace(node.PrivateKey)) < 1 ||
+					len(strings.TrimSpace(node.PublicKey)) < 1 {
+					errorList.PushBack("Field certs_by_ip for postgresql requires ip, private_key and public_key. Some of them are missing.")
+				}
+				errorList.PushBackList(checkCertValid([]keydetails{
+					{key: node.PrivateKey, certtype: "private_key", svc: "postgresql cert_by_ip for ip " + node.IP},
+					{key: node.PublicKey, certtype: "public_key", svc: "postgresql cert_by_ip for ip " + node.IP},
+				}))
+			}
+		} else {
+			if len(strings.TrimSpace(a.config.Postgresql.Config.RootCA)) < 1 ||
+				len(strings.TrimSpace(a.config.Postgresql.Config.PrivateKey)) < 1 ||
+				len(strings.TrimSpace(a.config.Postgresql.Config.PublicKey)) < 1 {
+				errorList.PushBack("Postgresql root_ca and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false.")
```

Comment:
```suggestion
				errorList.PushBack("PostgreSQL root_ca, public_key and private_key are mandatory fields, check if any of them are missing. Otherwise set enable_custom_certs to false.")
```